client/src/components/Profile/ProfileTabSelector/index.tsx

```diff
@@ -14,15 +16,18 @@ interface ProfileTabSelectorProps {
     setActiveTab: (tab: string) => void
 }
 
-export const ProfileTabSelector = (props: ProfileTabSelectorProps) => {
+export const ProfileTabSelector = observer((props: ProfileTabSelectorProps) => {
     const { activeTab, setActiveTab } = props
+    const navigate = useNavigate()
 
-    const tabSwitcher = (tab: string): void => {
+    const tabSwitcher = async (tab: string): Promise<void> => {
         setActiveTab(tab)
         if (tab === tabs.logOut) {
             setActiveTab(tabs.logOut)
-            userStore.setUser(null)
-            userStore.setIsAuth(false)
+            await localStorage.removeItem('token')
+            await userStore.setUser(null)
+            await userStore.setIsAuth(false)
+            navigate('/')
         }
```

Comment:
right now your tabSwitcher has two logics inside of it - switching tabs and also logging out functionality. I think (maybe, decide for yourself) it will be better to move this async logic with logging out to useEffect with activeTab dependency and you are going to have that if (tab === tabs.logOut) then you run that async logic

Reply:
Having two logics inside of one function is against S principle in SOLID